raiden/utils/__init__.py

```diff
@@ -288,3 +288,15 @@ def safe_gas_limit(*estimates) -> int:
     """
     calculated_limit = max(gas or 0 for gas in estimates)
     return int(calculated_limit * constants.GAS_FACTOR)
+
+
+def spawn_and_link_with_parent(func, *args, **kwargs):
+    """ Spawn a new greenlet and link it to the current thread instead of the hub. """
+    parent = gevent.getcurrent()
```

Comment:
The approach here is not wrong, but may have unexpected side effects.
Notice this will link erros on the spawned task with the current greenlet, which may be as far as the `_handle_response` greenlet from `GMatrixClient`. Imagine this greenlet being killed in the future (after handling of the event was already finished) because something bad happened on the transaction.
IMO, spawning and linking like that is the right thing to do (we do something similar [in matrix](https://github.com/raiden-network/raiden/blob/6b913826fcc5da01fc5b2d6c73cdf1c89fb1b0fd/raiden/network/transport/matrix.py#L437-L448), but it'd be better to link to a very explicit and specific runnable/greenlet/task, instead of the current one. E.g. the `handle_transaction_event` should better link with the `AlarmTask` runnable, even if it was actually called/processed in another greenlet, `AlarmTask` is the _right one_ to die and traceback on transaction errors.
Other than that, it seems right to do it.

Reply:
Initially I did tend to agree, but not anymore:
Some call chains do further exception handling, others don't: 

E.g. [calls through the API](https://github.com/raiden-network/raiden/blob/023554420534eb2ea8a1705bb7653e9f44d7d214/raiden/api/rest.py#L1103-L1109) might catch and handle `InsufficientFunds`, and are hence the right one's to receive the `on_error` exception.

Other's are expected to die on triggering `RaidenUnrecoverableError`.

If we are worried that some of our `Runnable`s might silently die (without killing the whole app), we should make sure that those are properly monitored/do the right exception escalation in their `on_error` handling.

Reply:
@andrevmatos is it necessary for `_handle_thread` to be short lived? This just won't work if that is the case.

Reply:
> yes, unfortunately, as we need to wait/block on it before starting the next /sync loop (so /sync calls don't overlap).

If this is the reason, why not just doing a blocking call?

Another solution is to return an async result from `handle_transaction_event` and wait on all of them in the functions `_initialize_transactions_queues` and `handle_state_change`

Reply:
@hackaugusto we avoid the blocking call to do our best to minimize time without a pending `/sync` request, since it keeps the online status on server, plus preventing the issue with overlapping calls.
For me, it still doesn't make sense to kill transport sync/handle thread because an async transaction Gosh knows how many levels down the stack failed. There should be a better place to re-raise/handle this class of exceptions.

Reply:
ping